### DISCUSSION TITLE
Lesson 6: ***forge script script/DeploySimpleStorage.s.sol***

### DISCUSSION THREAD

**Discussion by MarcoW3** [Cyfrin/foundry-full-course-cu#961] [Q&A]
Hi guys, when I run **the command in the title**, I receive this answer:

```
No files changed, compilation skipped
The application panicked (crashed).
Message:  no bytecode for contract
Location: crates/forge/bin/cmd/script/executor.rs:47

This is a bug. Consider reporting it at https://github.com/foundry-rs/foundry

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
Aborted
```

Thanks for helping!

----

Here my codes:

<details><summary>DeploySimpleStorage.s.sol</summary>
<p>

```
// SPDX-License-Identifier: MIT

pragma solidity 0.8.18;

import {Script} from "forge-std/Script.sol";
import {SimpleStorage} from "../src/SimpleStorage.sol";

contract DeploySimpleStorage is Script {
   function run() external returns(SimpleStorage){
    vm.startBroadcast();
    SimpleStorage simpleStorage = new SimpleStorage();
    vm.stopBroadcast();
    return simpleStorage;
   }
}
```
</p>
</details> 

<details><summary>SimpleStorage.sol</summary>
<p>

```
// SPDX-License-Identifier: MIT

pragma solidity 0.8.18;

// pragma solidity ^0.8.0;
// pragma solidity >=0.8.0 <0.9.0;

contract SimpleStorage {
    uint256 myFavoriteNumber;

    struct Person {
        uint256 favoriteNumber;
        string name;
    }
    // uint256[] public anArray;
    Person[] public listOfPeople;

    mapping(string => uint256) public nameToFavoriteNumber;

    function store(uint256 _favoriteNumber) public {
        myFavoriteNumber = _favoriteNumber;
    }

    function retrieve() public view returns (uint256) {
        return myFavoriteNumber;
    }

    function addPerson(string memory _name, uint256 _favoriteNumber) public {
        listOfPeople.push(Person(_favoriteNumber, _name));
        nameToFavoriteNumber[_name] = _favoriteNumber;
    }
}
```
</p>
</details>

**Comment by alfheimrShiven** [comment]
Hey @MarcoZ23 This seems to be a foundry-related bug. As advised by the error log, consider creating an issue at https://github.com/foundry-rs/foundry.

They should be able to give you some insight into what caused this.

**Reply by alfheimrShiven** [reply]
There is an already existing issue reported. Do follow the thread: https://github.com/foundry-rs/foundry/issues/3442

**Reply by alfheimrShiven** [reply]
Also try compiling after changing the `pragma solidity` version to `^0.8.18` in both your files. The thread suggests it's a version related issue.

**Reply by MarcoW3** [reply]
Thanks for the suggestions!

Actually, changing version to `^0.8.18` or higher doesn't change the returned answer.

I'm gonna report the bug (if it's so), and report here back any solution they give to me.

**Reply by alfheimrShiven** [reply]
Cool.

**Comment by MarcoW3** [comment]
https://github.com/foundry-rs/foundry/issues/6025#issuecomment-1760795882

Apparently, the bug disappeared by itself... 

Those guys tried with success. Then, me too...

But I'm curious now...
